controllers/v1/programs.js

```diff
@@ -3,13 +3,385 @@
-      return "programs";
+          let programUpdated = await programsHelper.removeRolesInScope(
+            req.params._id,
+            req.body.roles
+          );
+      
+          return resolve(programUpdated);
+  
+        } catch (error) {
+          return reject({
+            status: error.status || HTTP_STATUS_CODE.internal_server_error.status,
+            message: error.message || HTTP_STATUS_CODE.internal_server_error.message,
+            errorObject: error
+          });
+        }
+      });
+    }
+
+
+
+     /**
+    * @api {post} /unnati/v1/programs/removeEntitiesInScope/:programId Add roles in programs
+    * @apiVersion 1.0.0
+    * @apiName 
+    * @apiGroup Programs
+    * @apiParamExample {json} Request-Body:
+    * {
+      "entities" : ["5f33c3d85f637784791cd830"]
+    }
+    * @apiHeader {String} X-authenticated-user-token Authenticity token
+    * @apiSampleRequest /unnati/v1/programs/removeEntitiesInScope/5ffbf8909259097d48017bbf
+    * @apiUse successBody
+    * @apiUse errorBody
+    * @apiParamExample {json} Response:
+    * {
+        "message": "Successfully removed entities in program scope",
+        "status": 200
+      }
+    */
+
+     /**
+   * Remove entities in program scope
+   * @method
+   * @name removeEntitiesInScope
+   * @param {Object} req - requested data.
+   * @param {String} req.params._id - program id.
+   * @param {Array} req.body.entities - Entities to be added.
+   * @returns {Array} Program scope roles.
+   */
+
+  async removeEntitiesInScope(req) {
+    return new Promise(async (resolve, reject) => {
+      try {
+
+        let programUpdated = await programsHelper.removeEntitiesInScope(
+          req.params._id,
+          req.body.entities
+        );
+    
+        return resolve(programUpdated);
+
+      } catch (error) {
+        return reject({
+          status: error.status || HTTP_STATUS_CODE.internal_server_error.status,
+          message: error.message || HTTP_STATUS_CODE.internal_server_error.message,
+          errorObject: error
+        });
+      }
+    });
+  }
+
+
+
+   /**
+   * List programs.
+   * @method
+   * @name list
+   * @param {Object} req - Requested data.
+   * @param {Array} req.query.page - Page number.
+   * @param {Array} req.query.limit - Page Limit.
+   * @param {Array} req.query.search - Search text.
+   * @returns {JSON} List programs data.
+  */
+
+ async list(req) {
+  return new Promise(async (resolve, reject) => {
+    try {
+
+      let listOfPrograms = await programsHelper.list(
+        req.pageNo,
+        req.pageSize,
+        req.searchText
+      );
+
+      listOfPrograms["result"] = listOfPrograms.data;
+
+      return resolve(listOfPrograms);
+
+    } catch (error) {
+      return reject({
+        status: error.status || HTTP_STATUS_CODE.internal_server_error.status,
+        message: error.message || HTTP_STATUS_CODE.internal_server_error.message,
+        errorObject: error
+      });
     }
+  });
+}
+        
```

Comment:
@Prajwal17Tunerlabs programs/join api is missing in controller but it is added in helper